js/foam/ui/TableView.js

```diff
@@ -647,12 +648,18 @@ CLASS({
           self.mouseOverRow = false;
           self.selection = '';
         };
-        e.onclick = function(evt) {
-          self.hardSelection = self.selection = obj;
-          self.publish(self.CLICK, obj);
+        e.onclick = function() {
+          if (timer) window.clearTimeout(timer);
+          timer = window.setTimeout(function() {
+            console.log('onclick');
+            self.hardSelection = self.selection = obj;
+            self.publish(self.CLICK, obj);
```

Comment:
If you add an 'isMerged: delay', option to your single-click listener then
you can set a flag to cancel it from your double-click listener.

On 5 October 2015 at 11:55, adamvy <notifications@github.com> wrote:

> In js/foam/ui/TableView.js
> <https://github.com/foam-framework/foam/pull/416#discussion_r41161101>:
>
> > @@ -647,12 +648,18 @@ CLASS({
> >            self.mouseOverRow = false;
> >            self.selection = '';
> >          };
> > -        e.onclick = function(evt) {
> > -          self.hardSelection = self.selection = obj;
> > -          self.publish(self.CLICK, obj);
> > +        e.onclick = function() {
> > +          if (timer) window.clearTimeout(timer);
> > +          timer = window.setTimeout(function() {
> > +            console.log('onclick');
> > +            self.hardSelection = self.selection = obj;
> > +            self.publish(self.CLICK, obj);
>
> One problem with this approach is that javascript prevents you from doing
> certain things unless you're in a click handler, such a opening a window.
> If you add a timeout and someone is listening for tableview click events
> and trying to open popups for the rows they click on, it will no longer
> work.
>
> Can you do the timer work in your client code which is listening for table
> view click/doubleclick events?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/foam-framework/foam/pull/416/files#r41161101>.
>